apps/test-mcp-server/src/index.ts

```diff
@@ -249,7 +255,27 @@ async function main() {
       const httpReq = req as any;
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       const httpRes = res as any;
-      await transport.handleRequest(httpReq, httpRes, req.body);
+      let originalErr: unknown;
+      try {
+        await transport.handleRequest(httpReq, httpRes, req.body);
+      } catch (e) {
+        originalErr = e;
+      } finally {
+        if (!enableSessionManagement) {
+          const maybeClose = (
```

Comment:
@CharlieHelps it seems like `transport.close` is just always present, no need to check for it